### PR TITLE
Add blog templating overrides

### PR DESCRIPTION
**Pull request by 4kelly**
@squidfunk I wanted to throw you some love for all the work you've done with Material-Mkdocs. Hopefully this makes it a bit less tedious to post future blog articles. Maybe you can do a blog post about using this blog post template! 🤯 

Im not very experienced with front-end, so you'll probably have some suggestions to tidy this up!

I thought about parameterizing the authors, but decided to save it for a followup. 

> Known issues. I don't have insiders, so the "Was this page helpful" section at the bottom is missing.

<details>
    <summary>Old Blog Index</summary>
Screen Shot 2021-11-07 at 2 26 41 PM

</details>

<details>
    <summary>New Blog Index</summary>
  Screen Shot 2021-11-07 at 2 26 30 PM

</details>

<details>
    <summary>Old Blog Article</summary>
Screen Shot 2021-11-07 at 2 27 14 PM

</details>

<details>
    <summary>New Blog Article </summary>
Screen Shot 2021-11-07 at 2 27 05 PM
</details>

### CODE REVIEW

**Comment by squidfunk** [comment]
Many thanks for your work on this! I need to set aside some time to look into it.

**Comment by hellt** [comment]
thanks @4kelly !
I know it's a long shot, but it would be cool to have authors support, as mkdocs is often a doc page that is run by multiple people. For example:


```yaml
blog:
  dir: blog
  authors:
    - name: Ryan Kelly
      author_image: https://avatars.githubusercontent.com/u/93479011
      default: true
    - name: Roman
      author_image: https://avatars.githubusercontent.com/u/something
```

the default author is used for a blog post if name is not set (or not found), otherwise `blog_author: <name>` can be set in post meta to select the author

wdyt?

**Comment by 4kelly** [comment]
@hellt Definitely interested in something like this! 

My original goal was simply to eliminate some boilerplate for @squidfunk 's _personal_ blog here, rather than generalize it for everyone at this stage (as to keep the PR small and incremental). 

- However, looking more closely, I think this PR currently "pollutes" the public `python` release with these blog "v0" templates. Not sure if we can easily prevent that? Might need some help from @squidfunk.

So TL;DR, 
- If we can hide this from the `python` release, I think we can merge as is. Then make small incremental improvements in followup PRs. Maybe it becomes part of the theme, or maybe it's just for squid and is used as an example for others.
- if not, I'm happy to work on this PR to flesh it out for a broader audience.

**Comment by 4kelly** [comment]
We can also work out of the limelight in this repo https://github.com/4kelly/material-mkdocs-blog until we think we have something.

**Comment by squidfunk** [comment]
Just a note: proper blogging support will demand some plugin logic, e.g. for paging and index page generation. I also understand this PR as the reduction of some boilerplate, but there are some problems with the approach. I'll try to come back to it asap. I also may add a blog plugin at some point.

**Comment by 4kelly** [comment]
No rush or obligation at all @squidfunk.  

👍 Lets keep this scoped to reducing boilerplate for now if possible then.

**Comment by squidfunk** [comment]
Just as a side note – I've started working on a blogging plugin that will make this PR obsolete and won't make theme extension necessary at all 😊 However, thank you for your efforts!

**Comment by Andre601** [comment]
> Just as a side note – I've started working on a blogging plugin that will make this PR obsolete and won't make theme extension necessary at all 😊 However, thank you for your efforts!

Nice to hear that.

On a (very) personal note tho: Any chances that the blog thing could be in a lower insider tier? I'm aware that making mkdocs plugins isn't an easy task and that it takes time, but I personally feel like having it on a 12k tier, while the 5k one hasn't even been reached yet, is a bit... extreme.
Not to mention the sudden increase from 1k per tier to 2k per tier starting at 8k.

Again, this is more a personal thing, but for people like me who can't afford to join the Sponsors program is this extreme long waiting... Maybe have some 500 tiers in between, for some smaller stuff and to give people a better feeling of accomplishing something?

Anyway, ted-talk + rent: off

**Comment by squidfunk** [comment]
> On a (very) personal note tho: Any chances that the blog thing could be in a lower insider tier? I'm aware that making mkdocs plugins isn't an easy task and that it takes time, but I personally feel like having it on a 12k tier, while the 5k one hasn't even been reached yet, is a bit... extreme.

This is a deliberate calculation and not arbitrary. The effort to pull off a blog plugin that doesn't have the limitations of all existing blog plugins will take me a significant amount of time to develop and to integrate with all existing features. Same goes for the search – it took me multiple weeks to get this right, which is why it's on a higher tier.

Note that setting aside the time for developing the feature is only one part of the story. Maintenance, triaging issues, and fixing bugs even after the plugin is released are other things factored into that amount. If I decide to ship a feature, I decide to maintain it, so it will bind significant parts of my time in the future.

> Not to mention the sudden increase from 1k per tier to 2k per tier starting at 8k.

It's not that sudden. The first 6 tiers were spaced $500 apart. Most of those things were quite straight forward to develop, The next 6 tiers are spaced $1,000, as growth improved and funding goals were hit faster. In fact, the last $4,000 funding goal was hit in 45 days. The current one will take a little longer, but we'll get there. The $2,000 gap is just the logical progression of that pattern. Furthermore, the $2,000 funding goals all contain 5-6 features, whereas the lower goals only contained 3.

> Again, this is more a personal thing, but for people like me who can't afford to join the Sponsors program is this extreme long waiting... Maybe have some 500 tiers in between, for some smaller stuff and to give people a better feeling of accomplishing something?

Using Insiders is and always will be optional. Please don't think of Insiders as "I'm withholding something from you", but as "my private fork" where I develop new features I _personally_ find interesting and you can get access to that, but you don't have to. You can use Material for MkDocs without Insiders. You can use third-party blog plugins (which as I understand, you do), and put in your own time into getting it to work. The question is: how valuable is your own time to yourself? For example, sponsoring me for a year will cost you $120. That sounds like much, but it's actually less than a professional developer with the skillset that is necessary to pull something of the size and feature richness of Material for MkDocs will earn in a single hour. It's at least for me. So, in essence, with a yearly sponsorship, you're paying me less than 1 hour per year, which is fine and I'm happy to give you anything I develop for this amount because this is a mixed calculation, but I can't afford to do it for less or give stuff away for free. I also have to pay my bills and set aside time to work on this project.

If it weren't for the funding I would not be able to push this project forward. There would be no time to.

Please respect that.

**Comment by hellt** [comment]
> Just as a side note – I've started working on a blogging plugin that will make this PR obsolete and won't make theme extension necessary at all 😊 However, thank you for your efforts!

Awesome. @squidfunk do you have an estimated time of arrival? I'll prepare some champagne to crack open on a release date

**Comment by squidfunk** [comment]
@hellt judging from what I currently have on my plate probably around late spring. I need to finish work on [Instant Previews](https://twitter.com/squidfunk/status/1466794654213492743) and some smaller stuff first (including ongoing bugfixes and probably a little refactoring of the Mermaid integration when the funding goal is hit in the coming weeks).

**Comment by Andre601** [comment]
> > On a (very) personal note tho: Any chances that the blog thing could be in a lower insider tier? I'm aware that making mkdocs plugins isn't an easy task and that it takes time, but I personally feel like having it on a 12k tier, while the 5k one hasn't even been reached yet, is a bit... extreme.
> 
> This is a deliberate calculation and not arbitrary. The effort to pull off a blog plugin that doesn't have the limitations of all existing blog plugins will take me a significant amount of time to develop and to integrate with all existing features. Same goes for the search – it took me multiple weeks to get this right, which is why it's on a higher tier.
> 
> Note that setting aside the time for developing the feature is only one part of the story. Maintenance, triaging issues, and fixing bugs even after the plugin is released are other things factored into that amount. If I decide to ship a feature, I decide to maintain it, so it will bind significant parts of my time in the future.
> 
> > Not to mention the sudden increase from 1k per tier to 2k per tier starting at 8k.
> 
> It's not that sudden. The first 6 tiers were spaced $500 apart. Most of those things were quite straightforward to develop, and you have to start lower to motivate people to join the program, obviously. The next 6 tiers are spaced $1,000, as growth improved and funding goals were hit faster. In fact, the last $4,000 funding goal was hit in 45 days. The current one will take a little longer, but we'll get there. The $2,000 gap is just the logical progression of that pattern. Furthermore, the $2,000 funding goals all contain 5-6 features, whereas the lower goals only contained 3.
> 
> > Again, this is more a personal thing, but for people like me who can't afford to join the Sponsors program is this extreme long waiting... Maybe have some 500 tiers in between, for some smaller stuff and to give people a better feeling of accomplishing something?
> 
> Using Insiders is and always will be optional. Please don't think of Insiders as "I'm withholding something from you", but as "my private fork" where I develop new features I _personally_ find interesting and you can get access to that, but you don't have to. You can use Material for MkDocs without Insiders. You can use third-party blog plugins (which as I understand, you do), and put in your own time into getting it to work. The question is: how valuable is your own time to yourself? For example, sponsoring me for a year will cost you $120. That sounds like much, but it's actually less than a professional developer with the skillset that is necessary to pull something of the size and feature richness of Material for MkDocs will earn in a single hour. It's at least for me. So, in essence, with a yearly sponsorship, you're paying me less than 1 hour per year, which is fine and I'm happy to give you anything I develop for this amount because this is a mixed calculation, but I can't afford to do it for less or give stuff away for free. I also have to pay my bills and set aside time to work on this project.
> 
> If it weren't for the funding I would not be able to push this project forward. There would be no time to.
> 
> Please respect that.

Like I said is it a personal thing here. I don't mind you earning money and from your explanation does it make sense in the long run.

I just sometimes wish there was things like some kind of alternatives for the Insiders features.
Like a small explanation on how to add this feature yourself (if doable) through theme extension if you can't afford sponsoring...
But that's something for another day

**Comment by squidfunk** [comment]
Closing, as there's no intention to merge this for the reasons stated. Thanks again @4kelly for taking the time!
